public/Face.tsx

```diff
@@ -0,0 +1,86 @@
+import React from "react";
+import { faceToSvgString } from "../src/faceToSvgString";
+import { FaceConfig, Overrides } from "../src/types";
+import { objStringifyInOrder } from "./utils";
+
+/*
+    This component is responsible for rendering the face SVG string
+    It uses the faceToSvgString function to generate the SVG string
+    It requires the faceConfig to be set, with the overrides being optional
+    The width and className are also optional
+    The SVG string is then rendered in a div element
+    The SVG string is set as the innerHTML of the div element
+    
+    ***
+    Memoization is used to speed up rendering by reusing the same SVG string
+    ***
+*/
+
+export const memoizeWithDeepComparison = <Fn extends (...args: any[]) => any>(
+  fn: Fn,
+) => {
+  const cache = new Map<string, ReturnType<Fn>>();
+  return function (...args: Parameters<Fn>) {
+    const serializedArgs = args
+      .map((arg) =>
+        // objStringifyInOrder is used to stringify objects in a consistent order, flattening nested objects then sorting keys
+        typeof arg === "object"
+          ? objStringifyInOrder(arg)
+          : JSON.stringify(arg),
+      )
+      .join(",");
+    if (cache.has(serializedArgs)) {
+      return cache.get(serializedArgs);
+    }
+    const result = fn(...args);
+    cache.set(serializedArgs, result);
+    return result;
+  };
+};
```

Comment:
It would probably be better to use https://react.dev/reference/react/memo rather than this custom memoization function.

You might be able to just wrap Face in React.memo and have it work. If the objects passed as props to Face (such as faceConfig) are the same (in terms of identity, not necessarily a deep compare) then  it won't re-render. So then if you create a new faceConfig object if (and only if) you edit some property inside it, then React.memo will work with no custom compare function and also be fast.

But also, why is memoization useful here? Are we re-rendering the same exact face many times? Why?

Reply:
Memoization helps a bunch upon rending second, third, etc options on the page. Each face is built by copying the current FaceConfig from Zustand store, and running Override on it with the various Gallery options. 

So Memoization is helpful when you select skin color A, select skin color B, then go back to skin color A - we wouldn't need to do any calculations at that point. You can feel this especially impactful with sliders - if you slide all the way to the right, and then all the way back to original point, you can feel the rendering be a tick quicker on the way back

My worry about custom vs default memoization, and basically everywhere I use `objStringifyInOrder`, is the nested values being caught properly, being able to string-compare objs, and not having to worry about if we did a deep or shallow copy prior

I'm happy to test this a bit more, but ultimately went with the annoying custom memoization after running into problems originally

Reply:
In very unscientific testing I did, it doesn't seem faster to switch to a previously-selected option than to a new option. Are you seeing different behavior?

If we keep memoization, we should do something to avoid it staying in memory forever, in case someone is on the editor for a long time that'd be a memory leak.

But I think if we're able to lazy load the Face components, that might have a bigger impact than memoization, and might make memoization unnecessary. So I think that's the way to proceed... lazy load the rendering of Face components, and see if maybe that just solves all performance issues.

Reply:
Added lazy loading